plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -240,9 +247,31 @@ export class DeploymentManager {
     return await this.cache.readCache(getArtifactSpec(migration));
   }
 
-  async clone<C extends Contract>(address: string, args: any[], network?: string): Promise<C> {
+  /**
+   * Call an async function with a given amount of retries
+   * @param fn an async function that takes a signer as an argument. The function takes a signer
+   * because a new instance of a signer needs to be used on each retry
+   * @param retries the number of times to retry the function. Default is 5 retries
+   * @param timeLimit time limit before timeout in milliseconds
+   */
+  async asyncCallWithRetry(fn: (signer: SignerWithAddress) => Promise<any>, retries: number = 5, timeLimit?: number) {
+    const signer = await this.getSigner();
+    try {
+      return await asyncCallWithTimeout(fn(signer), timeLimit);
+    } catch (e) {
+      retries -= 1;
+      debug(`Retrying with retries left: ${retries}`);
+      debug(`Error is ${e}`);
+      if (retries === 0) throw e;
+      // XXX to be extra safe, we can also get the signer transaction count and figure out the next nonce
```

Comment:
This hasn't happened recently, but in the past, a hanging txn might actually get mined twice. This is worth investigating, but I haven't been able to replicate the issue with the most recent code.

Reply:
Ok good call out, something to pay attn to